server/src/main/scala/com/cloudera/livy/server/LivyServer.scala

```diff
@@ -19,40 +19,56 @@
 package com.cloudera.livy.server
 
 import java.io.{File, IOException}
-import java.net.InetAddress
-import java.util.EnumSet
+import java.util.{EnumSet, UUID}
+import java.util.concurrent.{ScheduledThreadPoolExecutor, TimeUnit}
 import javax.servlet._
 
+import org.apache.hadoop.security.{SecurityUtil, UserGroupInformation}
 import org.apache.hadoop.security.authentication.server._
-import org.apache.hadoop.security.SecurityUtil
 import org.eclipse.jetty.servlet.FilterHolder
 import org.scalatra.metrics.MetricsBootstrap
 import org.scalatra.metrics.MetricsSupportExtensions._
 import org.scalatra.servlet.{MultipartConfig, ServletApiImplicits}
 
 import com.cloudera.livy._
+import com.cloudera.livy.rsc.RSCConf
 import com.cloudera.livy.server.batch.BatchSessionServlet
 import com.cloudera.livy.server.interactive.InteractiveSessionServlet
 import com.cloudera.livy.util.LineBufferedProcess
 
+
 class LivyServer extends Logging {
 
   private val ENVIRONMENT = LivyConf.Entry("livy.environment", "production")
   private val SERVER_HOST = LivyConf.Entry("livy.server.host", "0.0.0.0")
   private val SERVER_PORT = LivyConf.Entry("livy.server.port", 8998)
-  private val AUTH_TYPE = LivyConf.Entry("livy.server.auth.type", null)
-  private val KERBEROS_PRINCIPAL = LivyConf.Entry("livy.server.auth.kerberos.principal", null)
-  private val KERBEROS_KEYTAB = LivyConf.Entry("livy.server.auth.kerberos.keytab", null)
-  private val KERBEROS_NAME_RULES = LivyConf.Entry("livy.server.auth.kerberos.name_rules",
+  private val AUTH_KERBEROS_PRINCIPAL = LivyConf.Entry("livy.server.auth.kerberos.principal", null)
+  private val AUTH_KERBEROS_KEYTAB = LivyConf.Entry("livy.server.auth.kerberos.keytab", null)
+  private val AUTH_KERBEROS_NAME_RULES = LivyConf.Entry("livy.server.auth.kerberos.name_rules",
     "DEFAULT")
+  private val AUTH_TYPE = LivyConf.Entry("livy.server.auth.type", null)
+
+  private val LAUNCH_KERBEROS_PRINCIPAL =
+    LivyConf.Entry("livy.server.launch.kerberos.principal", null)
+  private val LAUNCH_KERBEROS_KEYTAB =
+    LivyConf.Entry("livy.server.launch.kerberos.keytab", null)
+  private val LAUNCH_KERBEROS_REFRESH_INTERVAL =
+    LivyConf.Entry("livy.server.launch.kerberos.refresh_interval_seconds", 3600)
+  private val KINIT_FAIL_THRESHOLD =
+      LivyConf.Entry("livy.server.launch.kerberos.kinit_fail_threshold", 5)
+  private val KINIT_CACHE_ROOT_DIR =
+    LivyConf.Entry("livy.server.launch.kerberos.kinit_cache_root_dir", System.getenv("java.io.tmp"))
```

Comment:
There's no env variable with that name. You might be thinking of the system property named "java.io.tmpdir". Instead of this, I'd rather use `java.nio.file.Files.createTempDirectory` if this is not explicitly set. See `RSCDriver.java` for an example.